docs/DeveloperGuide.md

```diff
@@ -1,34 +1,247 @@
+
+
+![find seq diagram](https://imgur.com/Icg5rdB.png)
+### 3.7 Filter Feature
+This feature accepts multiple space-separated keywords to search for activities with tags matching each keyword.
+
+#### 3.7.1 Current Implementation
+* This feature is called by the user when the “filter” command is entered into the command line. The space separated strings following the command are the keywords to match activity tags with.
+* The Parser will create a FilterCommand object.
+* The FindCommand will invoke its own executeCommand() method.
+* The Parser's lastShownList will be cleared.
+* For each keyword:
+    * Then it will loop through activityList to find activities with tags that contain the keyword.
+    * If one is found, it will be added to lastShownList.
+    * printResults() method of the Ui will be called
+        * If lastShownList is not empty, it will print the matching activities.
+        * Else, it will respond to the user that there are no tasks which match the given keyword.
+
+
+![filter seq diagram](https://imgur.com/hybT3R9.png)
+## 4. Appendix
+### Product Scope
+#### Target user profile
+
+* Students with time management issues/poor time management skills.
+* Users who are reasonably comfortable using CLI apps.
+
+#### Value proposition
+
+Allow users to record their daily activities and track their time usage in a user-friendly manner. 
+
+### User Stories
 
 |Version| As a ... | I want to ... | So that I can ...|
 |--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+|v1.0|user|start a new activity|track the time spent on the activity|
+|v1.0|user|end an activity|track the total time i spent on the activity|
+|v1.0|user|abort an activity|record a different activity|
+|v1.0|user|store the completed activities in a list|analyse how I spent my time|
+|v2.0|user|delete an activity|remove activities that I do not want to track|
+|v2.0|user|search by activity name|view similar activities|
+|v2.0|user|filter activities by tags|view activities of the same category|
+|v2.0|user|view the activities by date|see how much time I have spent on different activities|
+|v2.0|user|continue my activities at another time|do other things between activities|
+|v2.0|user|edit past activities|keep a more accurate record of activities|
+|v2.0|user|automate the deletion of old activities|keep a more concise log of activities|
+
 
-## Non-Functional Requirements
+### Non-Functional Requirements
 
 {Give non-functional requirements}
 
-## Glossary
+### Glossary
 
 * *glossary item* - Definition
 
-## Instructions for Manual Testing
-
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+### Instructions for Manual Testing  
+  
+  #### Launch and Shutdown
```

Comment:
The Launch and Shutdown is not formatted properly. Perhaps, you have added in an extra space?